lib/configParser.ts

```diff
@@ -0,0 +1,230 @@
+import * as path from 'path';
+import * as glob from 'glob';
+import * as logger from './logger'
+
+// Coffee is required here to enable config files written in coffee-script.
+try {
+  require('coffee-script').register();
+} catch (e) {
+  // Intentionally blank - ignore if coffee-script is not available.
+}
+
+// LiveScript is required here to enable config files written in LiveScript.
+try {
+  require('LiveScript');
+} catch (e) {
+  // Intentionally blank - ignore if LiveScript is not available.
+}
+
+export class ConfigParser {
+  public constructor() {}
+
+  // Default configuration.
+  private config_ = {
```

Comment:
can there be an interface that defines the typeshape of valid configuration objects?